src/content/content-safari.js

```diff
@@ -25,56 +87,53 @@ let handleStateRequest = async function handleStateRequest (message) {
   received, would create an instance of a UI controller, following a lazy initialization approach.
    */
   if (!uiController) {
-    uiController = UIController.create(state, {
-      storageAdapter: LocalStorageArea,
-      app: { name: 'Safari App Extension', version: `${Package.version}.${Package.build}` }
-    })
-    uiController.state.setWatcher('panelStatus', sendStateToBackground)
-    uiController.state.setWatcher('tab', sendStateToBackground)
-
-    await uiController.init()
-
-    // A notification from a embedded lib that it is present on a page. Upon receiving this we should destroy all Alpheios objects.
-    document.body.addEventListener('Alpheios_Embedded_Response', () => {
-      // if we weren't already disabled, remember the current state
-      // and then deactivate before disabling
-      if (!uiController.state.isDisabled()) {
-        uiController.state.save()
-        if (uiController.state.isActive()) {
-          uiController.deactivate().catch((error) => console.error(`UI controller cannot be deactivated: ${error}`))
-        }
-      }
-      uiController.state.disable()
-    })
-  }
-
-  let requestState = TabScript.readObject(message.body)
-  let diff = uiController.state.diff(requestState)
-
-  if (diff.has('status')) {
-    if (diff.status === TabScript.statuses.script.ACTIVE) {
-      if (requestState.panelStatus) { uiController.state.panelStatus = requestState.panelStatus }
-      if (requestState.tab) { uiController.changeTab(requestState.tab) }
-      uiController.activate()
-        .then(() => sendStateToBackground('updateState'))
-        .catch((error) => console.error(`Cannot activate a UI controller: ${error}`))
-    } else if (uiController.isActivated && diff.status === TabScript.statuses.script.DEACTIVATED) {
-      uiController.deactivate()
-        .then(() => sendStateToBackground('updateState'))
-        .catch((error) => console.error(`UI controller cannot be deactivated: ${error}`))
+    if (diff.has('status') && diff.status === TabScript.statuses.script.ACTIVE) {
+      uiController = UIController.create(state, {
+        storageAdapter: LocalStorageArea,
+        app: { name: 'Safari App Extension', version: `${Package.version}.${Package.build}` }
+      })
+      await uiController.init()
+    } else {
+      // If uninitialized, ignore all other requests other than activate
+      // TODO: This is due to Safari background sending state request after reload. Need to fix
```

Comment:
is there still a TODO Here?

Reply:
Well, Safari is still sending request with "deactivated" state on some occasions (reloads? It's really hard to debug as it crashes often. It also seems sometimes that debugging in browser JS and extension's Swift code is not fully synchronized which is weird). I'm not sure if we can fix it or not.
So maybe I'll leave it as a reminder to return to it at some later time, if that's OK.